packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1210,7 +1223,8 @@ class WidgetsApp extends StatefulWidget {
     _debugShowWidgetInspectorOverrideNotifier.value = value;
   }
 
-  static final ValueNotifier<bool> _debugShowWidgetInspectorOverrideNotifier = ValueNotifier<bool>(false);
+  static ValueNotifier<bool> get _debugShowWidgetInspectorOverrideNotifier => _debugShowWidgetInspectorOverrideNotifierObject ??= ValueNotifier<bool>(false);
+  static ValueNotifier<bool>? _debugShowWidgetInspectorOverrideNotifierObject;
```

Comment:
It's kind of weird that this is in the app widget. I would expect to see this either in the binding, which is the singleton API multiplexer, or for it to be a per-app object (so if you have multiple App widgets, they each have their own).

Reply:
agree this singleton got put in the wrong place. WidgetBinding would be a better place that should already have the right lifecycle in tests.